
Status: I guess doko wouldn't like this change.  Let's keep it here for now..

--- gcc-4.0-4.0.old/debian/rules.defs	2004-09-23 13:10:05.000000000 +0200
+++ gcc-4.0-4.0/debian/rules.defs	2004-09-23 13:28:27.000000000 +0200
@@ -193,14 +193,10 @@
   with_java := yes
 endif
 
-java_no_systems := hurd-i386 kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
+java_no_systems := gnu kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
 ifneq (, $(filter $(DEB_TARGET_GNU_SYSTEM),$(java_no_systems)))
   with_java := disabled for $(DEB_TARGET_GNU_SYSTEM)
 endif
-ifeq ($(DEB_TARGET_ARCH),hurd-i386)
-  java_no_systems += hurd-i386
-  with_java := disabled for architecure $(DEB_TARGET_GNU_CPU)
-endif
 ifdef DEB_CROSS
   with_java := disabled for cross compiler package
 endif
@@ -245,7 +241,7 @@
 ifeq ($(with_common_libs),yes)
   with_libffi := yes
   no_ffi_cpus :=
-  no_ffi_systems := hurd-i386 kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
+  no_ffi_systems := gnu kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
   ifneq ($(with_java),yes)
     ifneq (, $(filter $(DEB_TARGET_GNU_CPU),$(no_ffi_cpus)))
       with_libffi := disabled for architecure $(DEB_TARGET_GNU_CPU)
@@ -253,9 +249,6 @@
     ifneq (, $(filter $(DEB_TARGET_GNU_SYSTEM),$(no_ffi_systems)))
       with_libffi := disabled for $(DEB_TARGET_GNU_SYSTEM)
     endif
-    ifeq ($(DEB_TARGET_ARCH),hurd-i386)
-      with_libffi := disabled for architecure $(DEB_TARGET_ARCH)
-    endif
   endif
   ifdef DEB_CROSS
     with_libffi := disabled for cross compiler package
@@ -433,16 +426,13 @@
 
 with_libgnat := yes
 ada_no_cpus := arm m68k
-ada_no_systems := hurd-i386 kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
+ada_no_systems := gnu kfreebsd-gnu knetbsd-gnu netbsd-elf-gnu
 ifneq (, $(filter $(DEB_TARGET_GNU_CPU),$(ada_no_cpus)))
   with_ada := disabled for architecure $(DEB_TARGET_GNU_CPU)
 endif
 ifneq (, $(filter $(DEB_TARGET_GNU_SYSTEM),$(ada_no_systems)))
   with_ada := disabled for $(DEB_TARGET_GNU_SYSTEM)
 endif
-ifeq ($(DEB_TARGET_ARCH),hurd-i386)
-  with_ada := disabled for architecure $(DEB_TARGET_ARCH)
-endif
 ifeq (ada, $(findstring ada,$(WITHOUT_LANG)))
   with_ada := disabled by environment
 endif
@@ -516,7 +506,7 @@
 ifdef DEB_CROSS
   with_check := disabled for cross compiler package
 endif
-check_no_systems := hurd-i386 kfreebsd-gnu knetbsd-gnu
+check_no_systems := gnu kfreebsd-gnu knetbsd-gnu
 ifneq (, $(filter $(DEB_TARGET_GNU_SYSTEM),$(check_no_systems)))
   with_check := disabled for $(DEB_TARGET_GNU_SYSTEM)
 endif
@@ -524,9 +514,6 @@
 ifeq ($(DEB_TARGET_GNU_CPU), $(findstring $(DEB_TARGET_GNU_CPU),$(check_no_cpus)))
   with_check := disabled for architecure $(DEB_TARGET_GNU_CPU)
 endif
-ifeq ($(DEB_TARGET_ARCH),hurd-i386)
-  with_check := disabled for architecure $(DEB_TARGET_ARCH)
-endif
 ifneq ($(WITHOUT_CHECK),)
   with_check := disabled by environment
 endif
